lib/clients/index.js

```diff
@@ -0,0 +1,57 @@
+function MetricClients(backends) {
+  this.clients = [];
+  if (backends) {
+    this.init(backends);
+  }
+}
+
+/**
+ * Init clients for each of the backend specified in the given backends
+ *
+ * @param {Array} backends the backend configurations
+ */
+MetricClients.prototype.init = function(backends) {
+  var self = this;
+  if (backends) {
+    backends.forEach(function(backendConfig) {
+      var client = self.initClient(backendConfig);
+      if (client) {
+        self.clients.push(client);
+      }
+    });
+  }
+};
+
+/**
+ * Init a single client for the given backend
+ *
+ * @param {Object} backend
+ */
+MetricClients.prototype.initClient = function(backend) {
+  var type = backend.type;
+  var clientModule;
+  try  {
+    clientModule = require('./' + type);
```

Comment:
@grdryn did some test, it will throw error if `type` is `null` or `undefined`, but it will load the current module if it's empty string. So we should add defensive check on the value of `type`